apm-agent-plugins/apm-jms-plugin/src/test/java/co/elastic/apm/agent/jms/JmsInstrumentationIT.java

```diff
@@ -114,25 +113,25 @@ public void endTransaction() throws Exception {
     @Test
     public void testQueueSendReceiveOnTracedThread() throws Exception {
         final Queue queue = createQueue();
-        testQueueSendReceiveOnTracedThread(() -> brokerFacade.receive(queue), queue);
+        testQueueSendReceiveOnTracedThread(() -> brokerFacade.receive(queue, 2000), queue, false);
     }
 
     @Test
     public void testQueueSendReceiveNoWaitOnTracedThread() throws Exception {
         final Queue queue = createQueue();
-        testQueueSendReceiveOnTracedThread(() -> loopReceive(() -> brokerFacade.receiveNoWait(queue), 3000), queue);
+        testQueueSendReceiveOnTracedThread(() -> brokerFacade.receiveNoWait(queue), queue, true);
```

Comment:
why did you get rid of the retry in favor of sleeping once? retrying with a longer timeout seems more robust but should not increase test durations

Reply:
Unfortunately it didn't add robustness, it kept on failing. Besides, it meant I would get a non-deterministic number of receive spans/transactions, as I instrument `receiveNoWait`, which I would like to invoke exactly once per test. 
I have a suspicion this has something to do with the way ActiveMQ works by pushing messages to consumers, so I added this voodoo sleep to get the broker a chance to execute internal async stuff, if it has such, before creating the consumer. One last attempt would be to add a sleep between consumer creation and invocation of `receive`.
If those solve the issue, I am more than fine with adding 200-400 ms to our tests. Otherwise, we will just give up this test in ActiveMQ. We will still test the instrumentation of `receiveNoWait` in Artemis. So far I don't recall failures there.